### PR TITLE
[feat] Support for more than 2 segments in MMBT

### PR DESCRIPTION
**Pull request by apsdehal**
Summary:
Segment embedding other than first 2 dim are initialized from the average of first 2 embeddings

This is required for the case where there are 3 segments such as post, ocr, and features.

Differential Revision: D22970794

### CODE REVIEW

File: mmf/models/mmbt.py [hunk 1]
```diff
@@ -352,6 +352,43 @@ def forward(self, sample_list):
...
+                # text segments start from 0 which means modal will
+                # be last, otherwise, it is 0, which is already is
+                if max_id != max_segment:
+                    modal_token_type_ids = torch.full(
+                        (input_modal.size(0), 1),
+                        fill_value=max_segment,
+                        dtype=torch.long,
+                        device=input_modal.device,
+                    )
+
```
Comment (vedanuj) [inline mmf/models/mmbt.py:390]: This can probably be refactored to first determine the proper value of the modal segment, and then set it using `torch.full` once (instead of using torch.zeros, torch.ones for different use cases).

```
modal_token_type_ids = torch.full(
                        (input_modal.size(0), 1),
                        fill_value=max_segment,
                        dtype=torch.long,
                        device=input_modal.device,
                    )
```

File: mmf/models/mmbt.py [hunk 2]
```diff
@@ -352,6 +352,43 @@ def forward(self, sample_list):
         if self.config.use_modal_end_token:
             modal_end_token = sample_list.input_ids[:, -1].clone().detach()
 
+        if hasattr(sample_list, "modal_token_type_ids"):
+            modal_token_type_ids = sample_list.modal_token_type_ids
+        else:
+            modal_token_type_ids = torch.zeros(
+                (input_modal.size(0)),
```
Comment (vedanuj) [inline mmf/models/mmbt.py:359]: Will the shape of the segment be always Bx1?

Comment (apsdehal) [reply mmf/models/mmbt.py:359]: We are relying on the broadcasting here.

File: mmf/models/mmbt.py (hunk 1 above)
Comment (apsdehal) [reply mmf/models/mmbt.py:390]: Refactored. Thanks for the suggestion.

**Comment by apsdehal** [comment]
Original implementation of MMBT had a bug in which both modal and text were using 0 as segment ids. I am retraining the models on HM and will update the PR with the updated models as well. The rest of the content in the PR is ready for review.
